db-service/lib/SQLService.js

```diff
@@ -128,7 +128,14 @@ class SQLService extends DatabaseService {
     if (cds.env.features.stream_compat) {
       if (query._streaming) {
         this._changeToStreams(cqn.SELECT.columns, rows, true, true)
-        return rows.length ? { value: Object.values(rows[0])[0] } : undefined
+        if (!rows.length) return
+        
+        const result = rows[0]
+        let [key, val] = Object.entries(result)[0]
```

Comment:
Using `[0]` is not allowed if the object has potentially multiple properties. Would be required to get the key from the `cqn.elements` to ensure that the correct value is deleted and set as `value`.